packages/manager/apps/pci-instance/src/core/ShellRoutingSync.tsx

```diff
@@ -0,0 +1,18 @@
+import { ShellContext } from '@ovh-ux/manager-react-shell-client';
+import { useContext, useEffect } from 'react';
+import { useLocation } from 'react-router-dom';
+
+export default function ShellRoutingSync() {
+  const location = useLocation();
+  const { routing } = useContext(ShellContext).shell;
+
+  useEffect(() => {
+    routing.stopListenForHashChange();
+  }, [routing]);
+
+  useEffect(() => {
+    routing.onHashChange();
+  }, [location, routing]);
+
+  return null;
```

Comment:
It should be a hook since there is nothing to display

Reply:
Yep, I think so, but, for now that's the way all the `pci-**` projects are initialized.
I guess that there will be a reintegration phase in manager-components hooks very soon so, if you agree we could make the change once it will be done?

Reply:
Yes it's not a big issue but as mentioned below the pci projects initialization doesn't follow the guidelines on project structure so we have to make some changes anyways :
- queryClient is at the root, not in the api folder
- there is no utils folder (but I think it's useful too)
- there is no translation files on pci projects ?
- router component is not in the component folder but in the routing folder
- there is no core folder (which contains those files) but there is a hook folder 😉

Reply:
@chipp972, yep I agree with that 👍🏻

I used the recent migrations to initialize the project and I kept their folder/file structure.

However, I agree that this does not respect the conventions in this guideline, I will make the necessary changes.

I still have some questions:

- the `core` folder exists in all the `pci-**` projects that have been migrated, I suggest deleting it and moving the components to the `components` folder.
- regarding the `utils` folder, it would be smart IMO to me to create a dedicated folder that can contain various utility files and their associated tests, what do you think?
- regarding the translations, I have not added them yet because I do not have any at this stage. Do you think I should still add the json files?

Reply:
Sure, I'll add the JSON `fr_FR` translation file. 
An empty one will be OK ?